projects/soc-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/ProbationCaseSearch.kt

```diff
@@ -0,0 +1,48 @@
+package uk.gov.justice.digital.hmpps.service
+
+import org.springframework.data.jpa.domain.Specification
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.entity.*
+import uk.gov.justice.digital.hmpps.model.*
+import uk.gov.justice.digital.hmpps.model.Team
+
+@Service
+class ProbationCaseSearch(val personRepository: DetailRepository) {
+    fun find(request: SearchRequest): ProbationCases =
+        ProbationCases(personRepository.findAll(request.asSpecification()).map { it.toProbationCase() })
+
+    fun crns(crns: Set<String>): ProbationCases =
+        ProbationCases(personRepository.findByCrnIn(crns).map { it.toProbationCase() })
+}
+
+private fun SearchRequest.asSpecification(): Specification<DetailPerson> = listOfNotNull(
+    firstName?.let { matchesForename(it) },
+    surname?.let { matchesSurname(it) },
```

Comment:
Do we need to include aliases here?  Or does the search endpoint they're currently using not do that?